### PR TITLE
feat(dataframe): implement "DataFrame.extend"

### PR DESCRIPTION
**Pull request by AChenQ**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-28v2d3k 实现 DataFrame.extend 方法](https://app.clickup.com/t/28v2d3k)

File: graviti/dataframe/column/series.py [hunk 1]
```diff
@@ -43,20 +43,16 @@ class Series:
...
         name: Union[str, int, None] = None,
     ) -> None:
-        if schema is not None:
-            # TODO: missing schema processing
-            pass
-
-        self._data = data
-        self.name = name
-        self.schema = schema
+        raise NotImplementedError("Not support generating Series by __init__.")
```
Comment (linjiX) [inline graviti/dataframe/column/series.py:55]: seems unrelated with this commit?

File: graviti/dataframe/frame.py [hunk 2]
```diff
@@ -573,3 +576,11 @@ def extend(self, objs: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
         2   d.jpg    4      4
 
         """
+        if not isinstance(objs, self.__class__):
+            objs = self.__class__.from_pyarrow(pa.array(objs))
```
Comment (linjiX) [inline graviti/dataframe/frame.py:580]: create pa.array with specific type.

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 3]
```diff
@@ -573,3 +580,10 @@ def extend(self, objs: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
         2   d.jpg    4      4
 
         """
+        if not isinstance(objs, self.__class__):
```
Comment (linjiX) [inline graviti/dataframe/frame.py:583]: `objs` looks like more than one instance.
maybe `objs` -> `values`?

File: graviti/dataframe/frame.py [hunk 4]
```diff
@@ -573,3 +580,10 @@ def extend(self, objs: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
...
+            objs = self.__class__.from_pyarrow(pa.array(objs, self.schema.to_pyarrow()))
+        else:
+            if self.schema != objs.schema:
```
Comment (linjiX) [inline graviti/dataframe/frame.py:586]: Portex schema do not have `__eq__` method.
Also here checking the pyarrow type is enough

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 5]
```diff
@@ -575,6 +582,13 @@ def extend(self, objs: Union[Iterable[Dict[str, Any]], "DataFrame"]) -> None:
         2   d.jpg    4      4
 
         """
+        if not isinstance(values, self.__class__):
+            values = self.__class__.from_pyarrow(pa.array(values, self.schema.to_pyarrow()))
```
Comment (linjiX) [inline graviti/dataframe/frame.py:586]: `from_pyarrow` or `_from_pyarrow`?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

File: graviti/dataframe/frame.py [hunk 6]
```diff
@@ -265,6 +265,10 @@ def _get_repr_body(self, flatten_data: List[ColumnSeries]) -> List[List[str]]:
             lines.append(line)
         return lines
 
+    def _extend(self, objs: "DataFrame") -> None:
```
Comment (linjiX) [inline graviti/dataframe/frame.py:268]: `objs` -> `values`?

**Review by linjiX — CHANGES_REQUESTED** [review]

**Review by linjiX — APPROVED** [review]

**Pull request merged**